examples/mcp_demo/README.md

```diff
@@ -0,0 +1,56 @@
+# MCP + Weave Integration Example
+
+This example demonstrates a comprehensive integration between the Model Context Protocol (MCP) and Weave for tracing. It showcases how to instrument both the client and server components to capture detailed traces of their interactions.
+
+## Features
+
+This example demonstrates:
+
+- Tools: Functions that can be called by clients
+- Resources: Static and dynamic data that can be read by clients
+- Prompts: Templated messages for consistent interactions
+
+## Files
+
+- `example_server.py`: A demo MCP server with various tools, resources, and prompts
+- `example_client.py`: A client that connects to the server and provides a command-line interface to interact with all server capabilities
+
+## Setup
+
+1. Add OPENAI_API_KEY in an `.env` file.
+
+```bash
+touch examples/mcp_demo/.env
+```
+
+2. Install the required dependencies:
+
+```bash
+uv pip install -e ".[mcp]"
+```
+
+3. No API keys are required for running this example
+
+## Running the Example
+
+```bash
+uv run --no-refresh --frozen --no-sync --only-group "" --extra mcp examples/mcp_demo/example_client.py examples/mcp_demo/example_server.py
```

Comment:
this process did not work for me - I had to do some custom stuff to get it to work in terms of python envs.

Reply:
can you share yours?

Reply:
yeah, i just ran `python examples/mcp_demo/example_client.py examples/mcp_demo/example_server.py`. The uv command of `only-group` did not seem to exist and i think this works due to a potentially specific setup on the machine.

Reply:
`python examples/mcp_demo/example_client.py examples/mcp_demo/example_server.py` works for me too. Let's go with it. Updating readme.